pyomo/contrib/doe/doe.py

```diff
@@ -101,6 +101,8 @@ def __init__(
         """
 
         # parameters
+        if type(param_init) != dict:
```

Comment:
Maybe a better way to do this is to use:
```
if not isinstance(param_init, dict):
```

Reply:
In other parts of Pyomo, we use `Config*` objects to help with option validation, but for these cases, I think `isinstance` is appropriate.

That said, unless you really, really need this to be a `dict` subclass, I would recommend `isinstance(param_init, collections.Mapping)` (which allows for any mapping type - even if they don't inherit from `dict`).  Similarly, I would test against `collections.Sequence` and not `list` in `measurements.py` (which admits other sequences like `tuple`)